fbw-common/src/wasm/systems/systems/src/communications/amu.rs

```diff
@@ -0,0 +1,861 @@
+
+    is_power_supply_powered: bool,
+}
+
+impl Computer {
+    pub fn new_cpt_and_avncs(context: &mut InitContext) -> Self {
+        Self {
+            audio_card: AudioCard::new(context, 1),
+            _second_card: TypeCard::Calls(Calls::new()),
+            is_power_supply_powered: false,
+        }
+    }
+
+    pub fn new_fo_and_ovhd(context: &mut InitContext) -> Self {
+        Self {
+            audio_card: AudioCard::new(context, 2),
+            // Not used for now
+            _second_card: TypeCard::Bite(Bite::new()),
+            is_power_supply_powered: false,
+        }
+    }
+
+    pub fn update(&mut self, context: &UpdateContext, bus: &mut Vec<Arinc429Word<u32>>) {
+        self.audio_card
+            .update(context, bus, self.is_power_supply_powered);
+    }
+
+    pub fn get_mixed_audio_acp(&self) -> MixedAudio {
+        self.audio_card.get_mixed_audio_acp()
+    }
+
+    pub fn get_mixed_audio_acp3(&self) -> MixedAudio {
+        self.audio_card.get_mixed_audio_acp3()
+    }
+
+    pub fn get_transmission_table_acp(&self) -> u32 {
+        self.audio_card.get_transmission_table_acp()
+    }
+
+    pub fn get_transmission_table_acp3(&self) -> u32 {
+        self.audio_card.get_transmission_table_acp3()
+    }
+}
+
+impl SimulationElement for Computer {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.audio_card.accept(visitor);
+
+        visitor.visit(self);
+    }
+
+    fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+        self.is_power_supply_powered = buses.is_powered(ElectricalBusType::DirectCurrentEssential);
+    }
+}
+
+pub struct AudioCard {
+    bus_acp: Vec<Arinc429Word<u32>>,
+    acp: AudioControlPanel,
+    mixed_audio_acp: MixedAudio,
+    mixed_audio_acp3: MixedAudio,
+    transmission_table_acp: u32,
+    transmission_table_acp3: u32,
+}
+
+impl AudioCard {
+    pub fn new(context: &mut InitContext, id_acp: u32) -> Self {
+        Self {
+            bus_acp: Vec::new(),
+            acp: AudioControlPanel::new(context, id_acp, ElectricalBusType::DirectCurrentEssential),
+            mixed_audio_acp: Default::default(),
+            mixed_audio_acp3: Default::default(),
+            transmission_table_acp: 0,
+            transmission_table_acp3: 0,
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        bus_from_adaptation_card: &mut Vec<Arinc429Word<u32>>,
+        is_powered: bool,
+    ) {
+        if is_powered {
+            self.acp.update(context, &mut self.bus_acp);
+
+            AudioCard::decode_arinc_words_from_acp(
+                &mut self.bus_acp,
+                &mut self.mixed_audio_acp,
+                &mut self.transmission_table_acp,
+            );
+
+            AudioCard::decode_arinc_words_from_acp(
+                bus_from_adaptation_card,
+                &mut self.mixed_audio_acp3,
+                &mut self.transmission_table_acp3,
+            );
```

Comment:
`AudioCard::decode_arinc_words_from_acp` needs 4 parameters but you only pass 3!?

Reply:
Hold on, commited by mistake before going to bed. It's fixed locally :)